File: src/main/java/com/ts/advices/domain/service/AdviceService.java
package com.ts.advices.domain.service;

import com.ts.advices.domain.AdviceRepositoryManager;
import com.ts.advices.domain.model.Advice;
import com.ts.advices.domain.model.Author;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;

@Component
public class AdviceService {

    private AdviceRepositoryManager adviceRepositoryManager;

    @Autowired
    public AdviceService(AdviceRepositoryManager adviceRepositoryManager) {
        this.adviceRepositoryManager = adviceRepositoryManager;
    }

    public List<Advice> getAdvicesByUserName(String userName) {
        return adviceRepositoryManager.getAdvicesByUserName(userName);
    }

    public Advice getAdviceBy(String id){
        return adviceRepositoryManager.getAdviceBy(id);
    }

    public Author getAuthorByUserName(String userName) {
        return adviceRepositoryManager.getAuthorByUserName(userName);
    }

    public Advice getRandomAdvice() {
        return adviceRepositoryManager.getRandomAdvice();
    }

    @Transactional
    public void deleteAdviceBy(String id) {
        adviceRepositoryManager.deleteAdviceBy(id);
    }

    @Transactional
    public Long save(Advice advice) {
        return adviceRepositoryManager.save(advice);
    }
}

File: src/main/java/com/ts/advices/domain/service/AdviceService.kt
package com.ts.advices.domain.service

import com.ts.advices.domain.AdviceRepositoryManager
import com.ts.advices.domain.model.Advice
import com.ts.advices.domain.model.Author
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Component
import org.springframework.transaction.annotation.Transactional

@Component
class AdviceService @Autowired constructor(val adviceRepositoryManager: AdviceRepositoryManager){

    fun getAdvicesByUserName(userName: String):List<Advice>? = adviceRepositoryManager.getAdvicesByUserName(userName)

    fun getAdviceBy(id: String): Advice? = adviceRepositoryManager.getAdviceBy(id)

    fun getAuthorByUserName(userName: String): Author? = adviceRepositoryManager.getAuthorByUserName(userName)

    fun getRandomAdvice(): Advice? = adviceRepositoryManager.getRandomAdvice()

    @Transactional
    fun deleteAdviceBy(id: String) = adviceRepositoryManager.deleteAdviceBy(id)

    @Transactional
    fun save(advice: Advice): Long? = adviceRepositoryManager.save(advice)

}